requests/utils.py

```diff
@@ -358,13 +358,12 @@ def get_encoding_from_headers(headers):
 
 def stream_decode_response_unicode(iterator, r):
     """Stream decodes a iterator."""
+    encoding = r.encoding
 
     if r.encoding is None:
```

Comment:
Attribute accesses are slower than local variable look-ups though, so we might as well use `encoding` since it already exists.